MRN-Code/bookshelf-shield
# Add coins-validate to project.

Asana task: https://app.asana.com/0/8456727477974/88447712467936
- **Problem:** Running `npm install` in [MRN-Code/nodeapi](github.com/MRN-Code/nodeapi) fails. The error appears to come from bookshelf-shield, specifically that [precommit-hook](https://www.npmjs.com/package/precommit-hook) is installed as a dependency. Precommit-hook’s git-validate errors with “can’t run bin/install script.”
- **Solution:** Remove precommit-hoook, add [coins-validate](https://www.npmjs.com/package/coins-validate)
- **Testing:**
  1. clone this repo onto a machine running nodeapi
  2. Check out this branch (_enhancement/add-coins-validate_)
  3. `cd` into the cloned repo.
  4. Run `npm install`
  5. Run `npm link` (may need to run `sudo npm link` on a devcoin due to network storage setup)
  6. Change directories to _nodeapi_ (`cd /coins/nodeapi`)
  7. Blow away the node modules (`rm -rf node_modules`)
  8. Run `npm link bookshelf-shield`
  9. Run `npm install`
